### DISCUSSION TITLE
error in calculation dislocation density

### DISCUSSION THREAD

**Discussion by BettyWalzier** [mtex-toolbox/mtex#686] [Ask Anything]
Hi all，

I was using the GND code (here: https://1drv.ms/u/s!AuHZyiTiAulymlDHlmMmH0yrBCtL?e=UaktOO) to calculate the GND density of pure Cu. An example of using the GND Code is here: https://cochranec.github.io/notch/ZhouyaoLoad.html. However, when I call the GND function, namely the function **[disArray systems]=GND_auto(ebsd,4,[0.3],cubicType);**
an error comes out as follows:
![image](https://user-images.githubusercontent.com/76571114/104046025-be77a100-51df-11eb-9dda-11880117cd91.png)
**This code is working fine for FCC steel (data is here: https://1drv.ms/u/s!AuHZyiTiAulymlP4C7Q2YFUIzv3B?e=QcNJOH). But why did it make a mistake when calculating the GND density of FCC-Cu?**
The data of pure Copper is here: https://1drv.ms/u/s!AuHZyiTiAulymk8cl24fboZ5IJSM?e=WJfgB8.
Is there anyone who knows how to solve this problem？

Thanks in advance!
Best regards,
Betty

**Comment by kilir** [comment]
Hi,

have you tried asking the author of the wrapper (https://github.com/tskippon/GND) ?

It might be good to provide also what you did since as far as I can tell, there's no function `Calc_GND` in mtex, so must be something you have.

Also, here might be a good start: https://mtex-toolbox.github.io/GND.html

Cheers,
Rüdiger

**Reply by BettyWalzier** [reply]
Hi Rüdiger,

Sorry, I upload the wrong image. The error is as follows:
![image](https://user-images.githubusercontent.com/76571114/104068600-87b48180-5205-11eb-94d3-4557d59cafb9.png)
I have sent an email to the author, but until now there is no response.
It's very strange. Both of them are FCC crystals, but for copper (https://1drv.ms/u/s!AuHZyiTiAulymk8cl24fboZ5IJSM?e=WJfgB8), it does not work. For steel (https://1drv.ms/u/s!AuHZyiTiAulymlP4C7Q2YFUIzv3B?e=QcNJOH), it works well.
Do you know where is the problem?

Best regards,
Betty

**Comment by zmichels** [comment]
Hi Betty,

Without the a version of the function you are using (which is not part if the standard mtex toolbox distribution), nobody can do much to help.

Zach

**Reply by BettyWalzier** [reply]
Hi Zach,

Thank you for your reminder. I used Matlab 2019a and MTEX 5.3.1 as well as MTEX 5.5.0.

Best regards,
Betty

**Comment by zmichels** [comment]
Hi Betty,

Sure. But that function mentioned in the error is not part of MTEX. So probably somebody else wrote it, if you didn't(?). And since the rest of us only have the functions that come *with* mtex, we have no way to help troubleshoot your custom function that is giving the error. Perhaps including the code or contacting the person who wrote that particular function will help.

Cheers,
Zach

**Comment by zmichels** [comment]
“GND_auto()” is not part of MTEX, so we would need to see what that is before any chance to help troubleshoot.

**Reply by BettyWalzier** [reply]
Hi zmichels,

All of the files for GND density calculation is here: https://1drv.ms/u/s!AuHZyiTiAulymlDHlmMmH0yrBCtL?e=UaktOO. The file of “GND_auto()” is here: https://1drv.ms/u/s!AuHZyiTiAulymlSc4BT3WBONNwnv?e=72lfaw.
You can download it and see the details.

Thanks in advance!
Best regards,
Betty

**Reply by BettyWalzier** [reply]
Hi zmichels,

You can use the data of FCC-copper (https://1drv.ms/u/s!AuHZyiTiAulymk8cl24fboZ5IJSM?e=WJfgB8.) to test the code.

Best regards,
Betty
